src/stylesheet-registry.js

```diff
@@ -212,6 +213,11 @@ export default class StyleSheetRegistry {
   }
 }
 
+export const globalStyleSheetRegistry = new StyleSheetRegistry()
```

Comment:
Unfortunately when the constructor of `StyleSheetRegistry` creates a new StyleSheet and injects it into the page. In the case when you don't use the global registry this would be useless.

Perhaps we can create a default and shared `StyleSheetRegistry` in `style.js` only when `this.context` is undefined?

Reply:
Two things:

**First thing:**
The predominant use case for this will be server rendering, where the sheet **isn't** injected (since it's on the server). When client rendering, we can just use the global instance.

**Second thing:**
Are you sure about that assessment? Either way, the `StyleSheet` gets injected when `StyleSheetRegistry` is constructed. If that happens only during SSR, it's a no-op, but if you're using `StyleSheetRegistryContext` as part of CSR, you'll still presumably construct a new `StyleSheetRegistry` and it will be injected into the DOM.

Reply:
> Either way, the StyleSheet gets injected when StyleSheetRegistry is constructed.

Correct

> but if you're using StyleSheetRegistryContext as part of CSR

If you used multiple `StyleSheetRegistry` each using a dedicated StyleSheet, you'll have the same amount of StyleSheets on the client (due to hydration the same code that ran on the server will run on the client).

What I meant with my previous comment is that if you don't use the `globalStyleSheetRegistry` instance, on the client you'll have one extra (unused) <style> element. 

Probably not a big deal and the following is not worth and ugly, but the above could be avoided if in `style.js` you do the following:

```jsx
let globalStyleSheetRegistry = null
export default class JSXStyle extends Component {
  // ... 
  get styleSheetRegistry() {
      if (this.context instanceof StyleSheetRegistry) { 
          return this.context
      }
      
       if (!globalStyleSheetRegistry) {
              globalStyleSheetRegistry = new StyleSheetRegistry()
       }
       return globalStyleSheetRegistry
   }
}
```

with 

```js
export const StyleSheetRegistryContext = React.createContext(null)
```

Reply:
Oh, I see what you mean.

Personally, I think having the global registry just always present is fine even if it's empty (especially since I suspect the predominant use case will only be wrapping it for a SSR then using the global registry for everything on the client). My NextJS example that I added doesn't suffer from this for exactly that reason (the _document code only runs on the server).

If you'd really like me to change something, let me know I can do it. But like I said, I think the current behavior in this PR is fine.